debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/EventCache.java

```diff
@@ -0,0 +1,167 @@
+                break;
+            }
+            currentQueueSegment = new QueueSegment(dataDir, currentQueueSegmentIndex,
+                    getFilePathWithIndex(currentQueueSegmentIndex));
+        }
+        LOGGER.info("Recovered {} events into event cache", hashCache.size());
+    }
+
+    private String getFilePathWithIndex(long index) {
+        String queueSegmentFileName = String.format("%s.%d.%s", QUEUE_SEGMENT_FILE_NAME, index,
+                QUEUE_SEGMENT_FILE_EXTENSION);
+        return Path.of(dataDir, QUEUE_FILE_DIR, queueSegmentFileName).toString();
+    }
+
+    // Duplicate in EventQueue.java
+    private void recoverLatestQueueSegment() {
+        // read dir to find all queue files
+        Path queueDirPath = Path.of(dataDir, QUEUE_FILE_DIR);
+        String searchGlob = String.format("%s.[0-9]*.%s", QUEUE_SEGMENT_FILE_NAME, QUEUE_SEGMENT_FILE_EXTENSION);
+        ArrayList<Path> filePaths = new ArrayList<>();
+        try {
+            DirectoryStream<Path> stream = Files.newDirectoryStream(queueDirPath, searchGlob);
+            for (Path entry : stream) {
+                filePaths.add(entry);
+            }
+            if (filePaths.size() == 0) {
+                // no files found. nothing to recover.
+                LOGGER.info("No files found matching {}. Nothing to recover", searchGlob);
+                return;
+            }
+            // extract max index of all files
+            long maxIndex = 0;
+            Path maxIndexPath = null;
+            for (Path p : filePaths) {
+                String filename = p.getFileName().toString();
+                String indexStr = filename.substring(QUEUE_SEGMENT_FILE_NAME.length() + 1,
+                        filename.length() - (QUEUE_SEGMENT_FILE_EXTENSION.length() + 1));
+                long index = Long.parseLong(indexStr);
+                if (index >= maxIndex) {
+                    maxIndex = index;
+                    maxIndexPath = p;
+                }
+            }
+            // create queue segment for last file segment
+            currentQueueSegmentIndex = maxIndex;
+            currentQueueSegment = new QueueSegment(dataDir, currentQueueSegmentIndex, maxIndexPath.toString());
+
+            LOGGER.info("Starting to recover event cache from queue segment-{} with byte count={}", maxIndexPath,
+                    currentQueueSegment.getByteCount());
+        } catch (IOException x) {
+            throw new RuntimeException(x);
+        }
+    }
+
+    public boolean isEventInCache(String event) {
+        return hashCache.contains(event.hashCode());
+    }
+
+    public void addEventToCache(String event) {
+        // Check if cache is full. If full, remove oldest event from cache and add new
+        // event
+        if (hashCache.size() >= maxCacheSize) {
+            LOGGER.info("Event cache is full. Removing oldest event from cache");
+            String oldestEvent = eventQueue.removeLast();
```

Comment:
I hope this is O(1) method. Confirm this for the specific LinkedList type you're using.